spring-integration-jms/src/main/java/org/springframework/integration/jms/dsl/JmsInboundGatewaySpec.java

```diff
@@ -137,6 +141,44 @@ public S destinationResolver(DestinationResolver destinationResolver) {
 		return _this();
 	}
 
+
+	/**
+	 * Set a SpEL expression to resolve a 'replyTo' destination from a request {@link jakarta.jms.Message}
+	 * as a root evaluation object if {@link jakarta.jms.Message#getJMSReplyTo()} is null.
+	 * @param replyToExpression the SpEL expression for 'replyTo' destination.
+	 * @return the spec.
+	 * @since 6.1
+	 * @see ChannelPublishingJmsMessageListener#setReplyToExpression(Expression)
+	 */
+	public S replyToExpression(String replyToExpression) {
+		return replyToExpression(PARSER.parseExpression(replyToExpression));
+	}
+
+	/**
+	 * Set a function to resolve a 'replyTo' destination from a request {@link jakarta.jms.Message}
+	 * as a root evaluation object if {@link jakarta.jms.Message#getJMSReplyTo()} is null.
+	 * @param replyToFunction the function for 'replyTo' destination.
+	 * @return the spec.
+	 * @since 6.1
+	 * @see ChannelPublishingJmsMessageListener#setReplyToExpression(Expression)
+	 */
+	public S replyToFunction(CheckedFunction<Message, ?> replyToFunction) {
+		return replyToExpression(new FunctionExpression<>(replyToFunction.unchecked()));
+	}
```

Comment:
Why do we need this; why not just

```suggestion
	public S replyToFunction(Function<Message, ?> replyToFunction) {
		return replyToExpression(new FunctionExpression<>(replyToFunction));
	}
```

??

Reply:
Sure, but then `.replyToFunction(message -> message.getStringProperty("myReplyTo"))` is not going to be clean and end-user has to handle that `JMSException` himself which is not runtime one.

Reply:
So, you are assuming that the user will use a JMS operation to determine the destination? I guess that's reasonable given that the root object is a `Message`.

OK.